### PR TITLE
Tinder/state machine

### PR DESCRIPTION
**Pull request by OscarSequeiros**
I have added an example of Sprint with state machine DSL by Tinder.

States diagram:
image

### CODE REVIEW

File: kotlin-jvm/src/main/kotlin/playground/StateMachine.kt [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+package playground.statemachine
+
+import com.tinder.StateMachine
+import playground.shouldBe
+
+/**
+ * Tinder/StateMachine - A kotlin DSL for finite state machine
+ *
+ * - [Tinder/State machine: Github](https://github.com/Tinder/StateMachine)
```
Comment (jmfayard) [inline kotlin-jvm/src/main/kotlin/playground/StateMachine.kt:11]: Your state diagram is very nice. 
Doesn't it work in IntelliJ to add markdown syntax for images in the kdoc? That would be great. 

`States diagram:
image`

File: kotlin-jvm/src/main/kotlin/playground/StateMachine.kt [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        when (validTransition.sideEffect) {
+            SideEffect.UpdatedBacklog -> log("Backlog was updated.")
+            SideEffect.IdentifiedImpediments -> log("All impediments were identified.")
+            SideEffect.ExternalFeedback -> log("External feedback was received.")
+            SideEffect.InternalFeedback -> log("Internal feedback was received.")
+        }
+    }
+}
+
+fun log(message: String) {
```
Comment (jmfayard) [inline kotlin-jvm/src/main/kotlin/playground/StateMachine.kt:88]: Please make the function private

File: kotlin-jvm/src/main/kotlin/playground/StateMachine.kt [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    println(message)
```
Comment (jmfayard) [inline kotlin-jvm/src/main/kotlin/playground/StateMachine.kt:89]: You could test the side effects here by cheating a bit: 

```kotlin
private var messages = mutableListOf<String>()
private fun log(message: String) { messages += message }

fun main() {
   // ....
   messages shouldBe listOf(....)
}
```

**Review by jmfayard — APPROVED** [review]

**Review by jmfayard — APPROVED** [review]

File: kotlin-jvm/src/main/kotlin/playground/StateMachine.kt (hunk 1 above)
Comment (OscarSequeiros) [reply kotlin-jvm/src/main/kotlin/playground/StateMachine.kt:11]: I am not sure if there is a markdown for images but I have added the image link. 😉

File: kotlin-jvm/src/main/kotlin/playground/StateMachine.kt (hunk 3 above)
Comment (OscarSequeiros) [reply kotlin-jvm/src/main/kotlin/playground/StateMachine.kt]: Totally agree. Done. 😉

**Comment by jmfayard** [comment]
Merged, thanks a lot @OscarSequeiros !

**Pull request merged**